src/NetMQ/NetMQPoller.cs

```diff
@@ -225,7 +263,7 @@ public void Add([NotNull] NetMQTimer timer)
                 throw new ArgumentNullException(nameof(timer));
             CheckDisposed();
 
-            Run(() => m_timers.Add(timer));
+            RunAsync(() => m_timers.Add(timer));
```

Comment:
For completeness, shouldn't we have `AddAsync` methods that return tasks? These sync methods could call into them and discard the `Task`.

Ditto other sync methods such as `Remove`.

Reply:
I didn't run into any problems with the current add implementations...

BUT since you ask, it is *probably* possible to do 
```
poller.Add(socket);
await poller.RemoveAsync(socket);
socket.Dispose();
```

and still have issues?  Maybe?  It is possibly a race condition?